gui-lib/framework/private/text-port.rkt

```diff
@@ -538,50 +544,159 @@
             [(null? txts) (void)]
             [else 
              (define fst (car txts))
-             (define-values (str/snp style)
+
+             (define (markup->snip markup style framed?)
+               (let* ([text (new markup-text%)]
+                      [snip (new editor-snip% [editor text] [with-border? framed?])])
+                 (send snip use-style-background #t)
+                 (send text set-styles-sticky #f)
+                 (insert-markup markup text style #f)
+                 (send text lock #t)
+                 snip))
+
+             (define (insert-markup markup text style inline?)
+               ; FIXME: Mike doesn't know how to apply style here without destroying
+               ; the style set by, say, srcloc-markup
```

Comment:
Does it work to use a `style-delta%`?

Reply:
The `style` is passed in from the outside - I didn't create it.

Reply:
I think these are `(get-out-style-delta)` etc., suggesting that they already are style deltas, but I have a hard time understanding how the names get turned into style objects.

Reply:
Fixed this as per our e-mail discussion by calling `set-style` on the snip.